functions/models/model.py

```diff
@@ -47,13 +47,13 @@ class ModelMetadata:
     sampling_rate: int = SAMPLING_RATE
 
     @staticmethod
-    def from_firestore_event(data: Dict[str, Any]) -> "ModelMetadata":
+    def from_firestore_event(data: Dict[str, Any]) -> "Model":
```

Comment:
Just a semantic note, factories should probably use `@classmethod` instead of `@staticmethod`

Reply:
Just learnt something new today :) I'm going to do separate this in a future PR